plugins/modules/jenkins_node.py

```diff
@@ -164,6 +180,13 @@ def __init__(self, module):
         self.url = module.params['url']
         self.num_executors = module.params['num_executors']
         self.labels = module.params['labels']
+        self.offline_message = module.params['offline_message']  # type: str | None
+
+        if self.offline_message is not None:
+            self.offline_message = self.offline_message.rstrip("\n")
+
+            if self.state != "disabled":
+                self.module.fail_json("can not set offline message when state is not disabled")
```

Comment:
Either I got it completely wrong, or your statement in the doc section says exactly the opposite. From the docs I would imagine that the error happens when trying to set the message in a node that is already disabled. Here you are raising the error when the node is not disabled.

I am not familiar with the Jenkins API, so not sure. Please double check.

Reply:
No problem, I doubled checked and I think the doc is clear on two different behaviours

- Can't set offline message unless setting state to `disabled` - offline message doesn't make sense unless is offline - raises an error.
- Can't **change** offline message if node is already offline because it's set via `toggleOffline` inside `python-jenkins` and you can't toggle from offline to offline - raises a warning.

I don't think there's a change to make here.

Reply:
Actually, maybe we could disambiguate current and requested state in the docs - would that clear things up for you?

Reply:
Oh my, re-reading now and it is obviously right. Sorry.
I don't see the need to disambiguate in the docs, it was a matter of paying more attention to it. Thanks.